controllers/slicegateway/slicegateway.go

```diff
@@ -543,94 +573,115 @@ func (r *SliceGwReconciler) GetGwPodNameAndIP(ctx context.Context, sliceGw *kube
+			// return pod.Name, pod.Status.PodIP
 		}
 	}
 
-	return "", ""
+	return podNameList, podIPList
 }
 
-func isGatewayStatusChanged(slicegateway *kubeslicev1beta1.SliceGateway, podName string, podIP string, status *gwsidecar.GwStatus) bool {
-	return slicegateway.Status.PodName != podName ||
-		slicegateway.Status.PodIP != podIP ||
-		slicegateway.Status.LocalNsmIP != status.NsmStatus.LocalIP
+func isGatewayStatusChanged(ctx context.Context, slicegateway *kubeslicev1beta1.SliceGateway, podName string, podIP string, status *gwsidecar.GwStatus) bool {
+	log := logger.FromContext(ctx).WithValues("type", "SliceGw")
+	log.Info("values present in slicegw status", slicegateway.Status)
+	log.Info("values needs to updated in slicegw status", podName, podIP, status)
+
+	return !contains(slicegateway.Status.PodNames, podName) ||
+		!contains(slicegateway.Status.PodIPs, podIP) ||
+		!contains(slicegateway.Status.LocalNsmIPs, status.NsmStatus.LocalIP)
 }
 
 func (r *SliceGwReconciler) ReconcileGwPodStatus(ctx context.Context, slicegateway *kubeslicev1beta1.SliceGateway) (ctrl.Result, error, bool) {
 	log := logger.FromContext(ctx).WithValues("type", "SliceGw")
 	debugLog := log.V(1)
 
-	podName, podIP := r.GetGwPodNameAndIP(ctx, slicegateway)
-	if podName == "" || podIP == "" {
-		log.Info("Gw pod not available yet, requeuing")
+	podNames, podIPs := r.GetGwPodNameAndIP(ctx, slicegateway)
+	if len(podNames) == 0 || len(podIPs) == 0 {
+		log.Info("Gw pods not available yet, requeuing")
 		return ctrl.Result{RequeueAfter: 5 * time.Second}, nil, true
 	}
+	updatedNsmIPs := []string{}
+	toUpdate := false
+	for i := 0; i < len(podNames); i++ {
+		sidecarGrpcAddress := podIPs[i] + ":5000"
 
-	sidecarGrpcAddress := podIP + ":5000"
+		status, err := r.WorkerGWSidecarClient.GetStatus(ctx, sidecarGrpcAddress)
+		log.Info("Tunnel status ----------------------->","tunnel status",status.TunnelStatus)
+		if err != nil {
+			log.Error(err, "Unable to fetch gw status")
+			return ctrl.Result{RequeueAfter: 10 * time.Second}, nil, true
+		}
 
-	status, err := r.WorkerGWSidecarClient.GetStatus(ctx, sidecarGrpcAddress)
-	if err != nil {
-		log.Error(err, "Unable to fetch gw status")
-		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil, true
+		debugLog.Info("Got gw status", "result", status)
+		log.Info("before calling status changed", "pod names", slicegateway.Status.PodNames, "pod ips", slicegateway.Status.PodIPs, "pod nsmips", slicegateway.Status.LocalNsmIPs)
+		if isGatewayStatusChanged(ctx, slicegateway, podNames[i], podIPs[i], status) {
+			log.Info("gw status changed", "status change", podNames, podIPs)
+			log.Info("after calling status changed", "pod names", slicegateway.Status.PodNames, "pod ips", slicegateway.Status.PodIPs, "pod nsmips", slicegateway.Status.LocalNsmIPs)
+			toUpdate = true
+		}
+		if status.TunnelStatus.IntfName == ""{
+			log.Info("tun0 down updting pod names and ips <<<<<<<<<<<<<<<<<<<<<","podnames",podNames)
```

Comment:
Review all the new logs added. Remove the ones that are not needed. And properly classify if they have to be debug or info.